### PR TITLE
fclones: init at 0.14.0

### PR DESCRIPTION
**Pull request by cyounkins**
Adding myself to maintainers and adding fclones tool.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bobby285271** [comment]
If you wish to maintain some existing packages or add a new package, can you also do that in this PR?

**Comment by cyounkins** [comment]
I read "just add yourself to that list in a separate commit" in the manual and interpreted it as separate PRs. Adding the commit for `fclones` now.

**Comment by nh2** [comment]
@msfjarvis packaged `fclones` at https://github.com/msfjarvis/custom-nixpkgs/blob/ca9a34702d34953ce90dd836f297961646d31f15/pkgs/fclones/default.nix with a derivation that also addresses Darwin support.

@msfjarvis would you like to contribute the Darwin support here, and be co-maintainer?

**Comment by msfjarvis** [comment]
> @msfjarvis packaged `fclones` at https://github.com/msfjarvis/custom-nixpkgs/blob/ca9a34702d34953ce90dd836f297961646d31f15/pkgs/fclones/default.nix with a derivation that also addresses Darwin support.
> 
> @msfjarvis would you like to contribute the Darwin support here, and be co-maintainer?

I'd love to, @cyounkins would you accept a PR to your branch adding Darwin support or should I open a separate one once this lands?

File: pkgs/tools/misc/fclones/default.nix
Comment (figsoda) [inline pkgs/tools/misc/fclones/default.nix:11]:
```suggestion
  src = fetchFromGitHub {
    owner = "pkolaczk";
    repo = pname;
    rev = "v${version}";
    sha256 = "1ni5amy903cy822yhw070bcrrixrw2m1vr66q1h32bc98pyv4w05";
  };
```

File: pkgs/tools/misc/fclones/default.nix
Comment (figsoda) [inline pkgs/tools/misc/fclones/default.nix:15]: `udev` should probably be in `buildInputs`?

File: pkgs/tools/misc/fclones/default.nix
Comment (figsoda) [inline pkgs/tools/misc/fclones/default.nix:20]: not needed

```suggestion
```

File: pkgs/tools/misc/fclones/default.nix
Comment (figsoda) [inline pkgs/tools/misc/fclones/default.nix:29]:
```suggestion
    maintainers = with maintainers; [ cyounkins ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/fclones/default.nix
Comment (cyounkins) [reply pkgs/tools/misc/fclones/default.nix]: Changed. I need to better understand the difference...

File: pkgs/tools/misc/fclones/default.nix
Comment (cyounkins) [reply pkgs/tools/misc/fclones/default.nix]: I could swear I needed this at one point, but it seems to work fine without it so I'm probably mistaken.

File: pkgs/tools/misc/fclones/default.nix
Comment (figsoda) [reply pkgs/tools/misc/fclones/default.nix]: `buildInputs` is usually for stuff like libraries and `nativeBuildInputs` is usually for stuff like build tools

**Comment by cyounkins** [comment]
@figsoda thank you for the review. I have made the requested changes.

@msfjarvis I would find it easiest if you could open a new PR once this one lands. Please tag me.

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 136806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fclones</li>
  </ul>
</details>

File: pkgs/tools/misc/fclones/default.nix
Comment (figsoda) [inline pkgs/tools/misc/fclones/default.nix:30]:
```suggestion
```

**Review by figsoda — APPROVED** [review]
works on x86_64-linux

File: pkgs/tools/misc/fclones/default.nix
Comment (cyounkins) [reply pkgs/tools/misc/fclones/default.nix:30]: Is there a style guide I'm missing? If there is I'll adhere to it.

I thought it was good practice to have a newline at the end of a file due to POSIX standards - https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Comment (figsoda) [reply pkgs/tools/misc/fclones/default.nix:30]: that looks like an extra new line to me, github usually displays files without a newline differently

Comment (cyounkins) [reply pkgs/tools/misc/fclones/default.nix:30]: I find it remarkably confusing the way different editors present newlines: https://stackoverflow.com/questions/16222530/why-do-i-need-vim-in-binary-mode-for-noeol-to-work

Here's what hexdump thinks the end of the file looks like:
```
0000310 7d20 0a3b 0a7d 000a
0000317
```

So is that a NUL and then an LF or a 4-byte encoded LF? I don't know but I don't think it matters.

Comment (figsoda) [reply pkgs/tools/misc/fclones/default.nix:30]: fiddled with `hexdump` with `set binary noeol` for a while, i don't think `hexdump` show trailing newlines as hex numbers

this is without newline
```
0000310 7d20 0a3b 007d
0000315
```

this is with trailing newline
```
0000310 7d20 0a3b 0a7d
0000316
```

this is with an extra newline
```
0000310 7d20 0a3b 0a7d 000a
0000317
```

Comment (Artturin) [reply pkgs/tools/misc/fclones/default.nix:30]: nixpkgs-fmt can be used to format files https://github.com/nix-community/nixpkgs-fmt

Comment (cyounkins) [reply pkgs/tools/misc/fclones/default.nix:30]: @Artturin Thanks! I'll start using that.

`nixpkgs-fmt` doesn't care about any number of trailing newlines. I'm still confused on what editors are doing and if hexdump could be hiding some bytes from me, but in the interest of closing this out I'm pushing an update with no blank line at the end. I think we're good to go!

**Pull request merged**
